src/vm/internal/otable.cc

```diff
@@ -1,271 +1,148 @@
+//TODO: remove dependency on outbuffer_t here
+#ifndef TESTING
+int ObjectTable::showStatus(outbuffer_t *out, int verbose) {
+
+    std::stringstream ss;
+    if (verbose == 1) {
+        ss <<  "Object name hash table status:" << std::endl;
+        ss <<  "Object name hash table status:" << std::endl;
+        ss << "------------------------------" << std::endl;
+        ss << "Elements:        " << objects_.size() << std::endl;
+        ss << "Memory(bytes):     " << objects_.size() * sizeof(Value) << std::endl;
+        ss << "Bucket count:    " << objects_.bucket_count() << std::endl;
+        ss << "Load factor:     " << objects_.load_factor() << std::endl;
+
+        outbuf_add(out, ss.str().c_str());
+      }
 
-  user_obj_lookups++;
-  if (ob) {
-    user_obj_found++;
-  }
-  return (ob);
+    if (!verbose) {
+        ss << "Memory used(bytes):     " << objects_.size() * sizeof(Value) << std::endl;
+        outbuf_add(out, ss.str().c_str());
+    }
+    return objects_.size() * sizeof(Value);
 }
+#endif
 
-/*
- * Print stats, returns the total size of the object table.  All objects
- * are in table, so their size is included as well.
- */
-
-static char sbuf[100];
-
-int show_otable_status(outbuffer_t *out, int verbose) {
-  const auto object_hash_table_size = CONFIG_INT(__OBJECT_HASH_TABLE_SIZE__);
-
-  int starts;
-
-  if (verbose == 1) {
-    outbuf_add(out, "Object name hash table status:\n");
-    outbuf_add(out, "------------------------------\n");
-    sprintf(sbuf, "%10.2f", objs_in_table / static_cast<float>(object_hash_table_size));
-    outbuf_addv(out, "Average hash chain length:       %s\n", sbuf);
-    sprintf(sbuf, "%10.2f", static_cast<float>(obj_probes) / obj_searches);
-    outbuf_addv(out, "Average search length:           %s\n", sbuf);
-    outbuf_addv(out, "Internal lookups (succeeded):    %lu (%lu)\n",
-                obj_searches - user_obj_lookups, objs_found - user_obj_found);
-    outbuf_addv(out, "External lookups (succeeded):    %lu (%lu)\n", user_obj_lookups,
-                user_obj_found);
-  }
-  starts = static_cast<long>(object_hash_table_size) * sizeof(object_t *) +
-           objs_in_table * sizeof(object_t);
-
-  if (!verbose) {
-    outbuf_addv(out, "Obj table overhead:\t\t%8d %8d\n",
-                object_hash_table_size * sizeof(object_t *), starts);
-  }
-  return starts;
-}
+std::string basename(std::string s) {
+    
+    //remove all leading forward slashes from string.
+    auto it1 = s.begin();
+    for(; *it1 == '/' ;++it1);
+    s.erase(s.begin(),it1);
+    
+    //find # in the string and delete it and all subsequent characters
+    auto it2 = s.find('#');
+    if(it2 != std::string::npos )
+        s.erase( s.begin() + it2, s.end() );
+    
+    //remove all repetitions of .c at the end of the string
+    it1 = s.end() - 1;
+    for(;it1-1 != s.begin() && *it1 == 'c' && *(it1-1) == '.';it1 -= 2);
+    s.erase(it1+1, s.end() );
+    return s;
+}
```

Comment:
https://stackoverflow.com/questions/216823/whats-the-best-way-to-trim-stdstring 

i felt maybe we should move stuff in package/trim/trim.cc  out into a base/ as a new string utilities.